client/src/__tests__/unit-tests/bitbake-settings.test.ts

```diff
@@ -0,0 +1,63 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) 2023 Savoir-faire Linux. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import * as vscode from 'vscode'
+import { loadBitbakeSettings } from '../../driver/BitbakeSettings'
+
+jest.mock('vscode')
+
+const mockBitbakeConfiguration = (values: Record<string, string>): void => {
+  vscode.workspace.getConfiguration = jest.fn().mockImplementation(() => ({
+    get: (key?: string): string => {
+      if (key === undefined || values[key] === undefined) {
+        return ''
+      }
+      return values[key]
+    }
+  }))
+}
+
+const mockVsCodeWorkspace = (workspacePath: string): void => {
+  const workspaceFolder: vscode.WorkspaceFolder = {
+    uri: {
+      scheme: '',
+      authority: '',
+      path: '',
+      query: '',
+      fragment: '',
+      fsPath: workspacePath,
+      with: function (change: { scheme?: string | undefined, authority?: string | undefined, path?: string | undefined, query?: string | undefined, fragment?: string | undefined }): vscode.Uri {
+        throw new Error('Function not implemented.')
+      },
+      toJSON: function () {
+        throw new Error('Function not implemented.')
+      }
+    },
+    name: '',
+    index: 0
+  }
+
+  Object.defineProperty(vscode.workspace, 'workspaceFolders', {
```

Comment:
Nice mock here. It replaces the getter and spies on custom one so the expression in bitbakeSettings is got around.

Reply:
Thanks, that one was tricky to do :p.